libraries/ui-library/src/components/six-select/util.ts

```diff
@@ -0,0 +1,64 @@
+export interface MenuItem {
+  value: string;
+}
+
+export function getValue(value: unknown, multiple: boolean, menuItems: MenuItem[]): string | string[] {
+  if (multiple) {
+    if (Array.isArray(value)) {
+      return menuItems.filter((item) => value.includes(item.value)).map((item) => item.value);
+    } else if (typeof value === 'string') {
+      return menuItems.filter((item) => value === item.value).map((item) => item.value);
+    } else {
+      return [];
+    }
+  } else {
+    if (typeof value === 'string') {
+      return menuItems.find((item) => value === item.value)?.value ?? '';
+    } else {
+      return '';
+    }
+  }
+}
```

Comment:
This function is very nested which makes it hard to read. I would propose to at least unwrap the unnecessary `else` to increase code comprehension and code maintainability.

E.g.:

```typescript
export function getValue(value: unknown, multiple: boolean, menuItems: MenuItem[]): string | string[] {
    const mapToValue = (item) => item.value;

    if (multiple) {
        if (Array.isArray(value)) {
            return menuItems.filter((item) => value.includes(item.value)).map(mapToValue);
        } else if (typeof value === 'string') {
            return menuItems.filter((item) => value === item.value).map(mapToValue);
        }

        return [];
    }

    if (typeof value === 'string') {
        return menuItems.find((item) => value === item.value)?.value ?? '';
    }

    return '';
}
```

Reply:
Tried to simplify it a little: ccf3feb0fbf21f40a5202a2489eff028507c35b5